File: AI/js/classes/Bullet.js
class Bullet {
    constructor ({x, y, velocity, demage, direction, ID}) {
        this.demage = demage;
        this.ID = ID;
        this.vel = velocity;
        this.pos = createVector(x, y);
        this.direction = direction;
    }

    update () {
        let velocity = p5.Vector.fromAngle(this.direction, this.vel)
        this.pos.add(velocity);
    }

    show() {
        stroke(255);
        fill('rgba(0,255,0, 0.25)');
        ellipse(this.pos.x , this.pos.y, 10)
    }

}

File: AI/py/Bullet.py
from p5 import *
import numpy as np
from fromAngle import fromAngle

class Bullet:
    def __init__ (self, x, y, magni, demage, direction, ID):
        self.demage = demage
        self.ID = ID
        self.magni = magni
        self.vel = fromAngle(direction, magni)
        self.pos = np.array([x, y])
        self.direction = direction

    def update (self):
        self.pos = self.pos + self.vel


    def show(self) :
        stroke(255)
        fill(100)
        circle(self.pos[0] , self.pos[1], 10)